src/picologging/logrecord.cxx

```diff
@@ -137,6 +138,10 @@ int LogRecord_init(LogRecord *self, PyObject *initargs, PyObject *kwds)
     self->levelname = levelname;
     self->pathname = pathname;
     Py_INCREF(pathname);
+    self->process = getpid();
+
+    PyObject* sysModules = PySys_GetObject("modules"); // borrowed
+    PyObject* multiprocessing = PyDict_GetItemString(sysModules, "multiprocessing"); // borrowed
```

Comment:
I'm worried about the performance overhead of this, it would be good to benchmark before/after.

getpid() is relatively fast, so maybe a cache map between pid and the resolved name would be a fast solution. What do you think?

Reply:
I think that's good, but what happens if OS starts recycling the PIDs? We will be logging invalid process names and I can't think of how to invalidate it.
I think the best option would be to have a flag to be disabled by default False, and they can enable it to log process names. What do you think?